xmlutil.py

```diff
@@ -9,11 +9,13 @@ def append(text):
         f.write(text + '\n')
 
 
-def push(channel_name, data):
+def push(channel_name, data, icon):
     append('<channel id="{}">'.format(channel_name))
     append('<display-name>{}</display-name>'.format(escape(channel_name)))
+    if icon is not None:
+        append('<icon src="{}"/>'.format(escape(icon)))
```

Comment:
you can just do it here dynamically instead of copy/pasting urls onto the channels.py
`append('<icon src="{}"/>'.format(escape(f'https://data.greektv.ml/logos{channel_name}.png')))`

Reply:
Some Icons don't have the same name, like makedonia tv.

Reply:
and then we don't need any of the changes you did

Reply:
Also you are missing the icon for epsilon tv...

Reply:
> Some Icons don't have the same name, like makedonia tv.

Then just rename the epgName on channels.py

Reply:
> Also you are missing the icon for epsilon tv...

ae.png

Reply:
I am not sure if it's a good idea to change the names of the epgName because if anyone is using this xml will have issues and will have to setup it from scratch.
Also if I change the epsilon to ae, it's not recognizable.
Another problem by setting a default url for all the channels will make it difficult for someone who wants to fork this project to add custom icons.

Reply:
Ok fine